File: sdk/node/src/common/cache_utils.ts

export class CacheUtils {
    private static cache: { [key: string]: any } = {};

    static setCache(key: string, value: any): void {
        this.cache[key] = value;
        console.log(`Set cache for key: ${key}`);
    }

    static getCache(key: string): any | null {
        return this.cache[key] || null;
    }

    static loadCache(): { api_key: string | null, access_token: string | null } {
        return {
            api_key: this.getCache('api_key'),
            access_token: this.getCache('access_token')
        };
    }

    static saveCache(apiKey: string, accessToken: string): void {
        this.setCache('api_key', apiKey);
        this.setCache('access_token', accessToken);
        console.log('Saved API key and access token to cache.');
    }

    static clearCache(): void {
        this.cache = {};
        console.log('In-memory cache cleared.');
    }
}

export default CacheUtils;

File: sdk/python/src/common/cache_utils.py
class CacheUtils:
    _cache = {}  

    @staticmethod
    def set_cache(key, value):
        "Sets a key-value pair in the in-memory cache."
        CacheUtils._cache[key] = value
        print(f"Set cache for key: {key}")

    @staticmethod
    def get_cache(key):
        "Gets the value for a key from the in-memory cache."
        return CacheUtils._cache.get(key, None)

    @staticmethod
    def load_cache():
        "Loads API key and access token from the in-memory cache."
        return {'api_key': CacheUtils.get_cache('api_key'), 'access_token': CacheUtils.get_cache('access_token')}

    @staticmethod
    def save_cache(api_key, access_token):
        "Saves the current API key and access token to the in-memory cache."
        CacheUtils.set_cache('api_key', api_key)
        CacheUtils.set_cache('access_token', access_token)
        print("Saved API key and access token to cache.")

    @staticmethod
    def clear_cache():
        "Clears the in-memory cache."
        CacheUtils._cache.clear()
        print("In - memory cache cleared.")
